docs/development_guidelines.md

```diff
@@ -0,0 +1,45 @@
+# Development guidelines
+
+```{note}
+This document is a work in progress and ODIES is rapidly evolving; we'll continue to add to these guidelines as we develop the tool.
+```
+
+We've already had divergent codebases in the future due to fast-paced project needs and tight timelines.
+To make this better, we're trying to standardize the way we develop code for ODIES in a way that's beneficial for developers, does not slow progress, and is sustainable for the future.
+This document outlines the guidelines for developing code for ODIES
+
+## Test development
+
+Each new feature or bug fix should have a corresponding test.
+Additionally, to help users create subsystems with the correct methods and attributes, we should have tests that users can run to check their technology models.
+
+First, we will define different types of tests that we have in ODIES:
+- Unit tests: tests for individual functions or methods
+- Integration tests: tests for the interaction between different functions or methods
+- Regression tests: tests for ensuring that new changes do not break existing functionality
+
+### Test location
+
+Each folder within the `ODIES` package should have a corresponding `tests` folder.
```

Comment:
General comment that may not apply here: Should the unit, integration, and regression get separate folders within `tests/` that then roughly or directly mimic the code structure within? This then allows for easily setting up a workflow for running the each individually, such as `pytest --unit` (https://github.com/NREL/OpenOA/blob/main/test/conftest.py#L15-L37). 

I could certainly see `unit/` following this guidance on mimicking the structure, but regression and integration probably might be more focused on workflows than smaller units of testing.

Reply:
I think I prefer having the tests for each subpackage of the code living within that subpackage. So the tests wouldn't be off in their own folder with a mimicked structure, but instead right within the structure. Here's an example of that: https://github.com/OpenMDAO/OpenMDAO/tree/master/openmdao/core the core folder has files and then a `tests` folder that has unit tests for each of those files. What do you think about that kind of setup instead?

I agree that regression and integration would be separate in a higher level test folder.